doc/dev/error-categorization.md

```diff
@@ -0,0 +1,167 @@
+would require sudo permissions). Reactions are fixing the permissions or creating the file/directory
+and retry the operation.
+
+#### Installation ("C01200")
+
+`Installation Errors` are those errors which are related to a wrong installation such as
+wrong plugin names, missing backends, initialization errors, misconfiguration of Elektra etc.
+Installation errors might also be non-Elektra specific but also from dependent library/applications
+such as gpg.
+Also plugin configuration errors belong to `Installation Errors` as they this happens during
+mounting.
+Users will have to reconfigure, reinstall, recompile (with other settings) Elektra in order to
+get rid of this error or fix the installation of the corresponding library/application.
+
+#### Logical ("C01300")
+
+`Logical Errors` is another branch category in which you indicate a bug in Elektra
+such as internal errors, assertion failures or errors
+which should not happen such as going into a `default` branch when you are assured that all cases
+are covered. Usually such errors come with a message to report such failures to Elektra's bugtracker.
+Applications cannot handle such errors themselves.
+
+##### Assertion ("C01310")
+
+`Assertion Errors` are such errors which indicate a flaw in the internal logic such as going into a `default`
+branch which you do never expect to happen.
+This category might be used in the future to automatically
+issue a bug to our bugtracker. As of now if you have to use this error please add a message
+indicating that this bug should be reported.
+
+##### Interface ("C01320")
+
+`Interface Errors` errors indicate a wrong usage of Elektra's API. An example would be to pass a NULL pointer to
+`kdbGet`. Also violations of the backend belong into this category. Compared to semantic validation errors,
+this category has its focus on detecting wrong usages of the API instead of a "retry with a different value" approach.
+
+##### Plugin is Broken ("C01330")
+
+`Broken Plugin Errors` errors indicate that a plugin does not behave in an intended way. Unrecognized commands,
+unkown return codes, plugin creation errors, etc. belong to this category. Also uncaught exceptions belong here because
+Elektra expects all exceptions to be caught.
+
+### Conflicting State ("C02000")
+
+`Conflicting State Errors` are errors where the current state is incompatible with the attempted operation.
+Examples are the need for calling `kdbGet` before `kdbSet`. Another example would be to try to push you changes
+into a git repository where the remote branch has already changed.
+These kind of errors are usually in resolver plugins when the state of the file
+has changed without the system knowing. Such errors demand a special synchronization action before retrying.
+
+### Timeout ("C03000")
+
+`Timeout Errors` are easy to categorize as the reaction simply suggests to wait for a short period and retry again.
```

Comment:
All errors are easy to categorize when you know the category. (Please rewrite according to structure above)